hal-x86_64/src/vga.rs

```diff
@@ -146,8 +138,18 @@ impl Buffer {
             c.write(blank)
         }
 
-        self.col = self.indent;
-        self.chars_this_line = false;
+        self.col = 0;
+    }
+
+    fn clear(&mut self) {
+        let ch = self.character(b' ');
+        for row in self.buf.iter_mut() {
+            for col in row.iter_mut() {
+                col.write(ch);
+            }
+        }
```

Comment:
It might be more efficient to do a single volatile write to the entire VGA buffer, if that's allowed, like:

```rust
ptr::write_volatile(&mut *self.buf, [[self.character(b' '); BUF_WIDTH]; BUF_HEIGHT]);
```

Reply:
hmm, i wonder what that compiles to...